templates/file_viewer_template.html

```diff
@@ -110,40 +116,51 @@
-                            elem.includes(app_state.valid) && elem.includes('_'+app_state.model+'_') && elem.includes('_'+app_state.region+'_') &&
-                            elem.includes(app_state.plottype) && elem.includes(app_state.plotname) && elem.includes(app_state.fclt)) } );
-                } else if (app_state.selected == 'plottype') {
-                    imgs = imgs_all.filter( function(elem) { return (
-                            elem.includes(app_state.valid) && elem.includes('_'+app_state.model+'_') && elem.includes('_'+app_state.region+'_') &&
-                            elem.includes(app_state.timeagg) && elem.includes(app_state.plotname) && elem.includes(app_state.fclt)) } );
-                } else if (app_state.selected == 'plotname') {
-                    imgs = imgs_all.filter( function(elem) { return (
-                            elem.includes(app_state.valid) && elem.includes('_'+app_state.model+'_') && elem.includes('_'+app_state.region+'_') &&
-                            elem.includes(app_state.timeagg) && elem.includes(app_state.plottype) && elem.includes(app_state.fclt)) } );
-                } else if (app_state.selected == 'fclt') {
-                    imgs = imgs_all.filter( function(elem) { return (
-                            elem.includes(app_state.valid) && elem.includes('_'+app_state.model+'_') && elem.includes('_'+app_state.region+'_') &&
-                            elem.includes(app_state.timeagg) && elem.includes(app_state.plottype) && elem.includes(app_state.plotname)) } );
-                } else {
-                    // Otherwise, loop through the valid times
-                    imgs = imgs_all.filter( function(elem) { return (
-                            elem.includes('_'+app_state.model+'_') && elem.includes('_'+app_state.region+'_') && elem.includes(app_state.timeagg) &&
-                            elem.includes(app_state.plottype) && elem.includes(app_state.plotname) && elem.includes(app_state.fclt)) } );
+
+                // Normalise search criteria by mapping dimension to pattern
+                let tokens = {
+                    valid: app_state.valid,
+                    model: '_'+app_state.model+'_',
+                    region: '_'+app_state.region+'_',
+                    timeagg: app_state.timeagg,
+                    plottype: app_state.plottype,
+                    plotname: app_state.plotname,
+                    fclt: app_state.fclt
+                }
+
+                // Choose dimensions by plot name
+                let getImageDimensions = plotname => {
+                    // TODO: Add dimension logic here or use Jinja2??
+                    let nameToDims = {
+                        Foo: ["valid", "model", "region", "timeagg", "plottype", "plotname"],
+                        Bar: ["valid", "region", "timeagg", "plottype", "plotname"],
+                        Baz: ["valid", "model", "region", "plottype", "plotname"]
+                    }
```

Comment:
@claretandy the intention here is to have a mapping of special cases that could be injected by Jinja2 as a global variable